packages/api/cms-api/src/warnings/warning.resolver.ts

```diff
@@ -25,10 +30,73 @@ export class WarningResolver {
     }
 
     @Query(() => PaginatedWarnings)
-    async warnings(@Args() { status, search, filter, sort, offset, limit }: WarningsArgs): Promise<PaginatedWarnings> {
-        const where = gqlArgsToMikroOrmQuery({ search, filter }, this.repository);
+    async warnings(
+        @Args() { status, search, filter, scopes, sort, offset, limit }: WarningsArgs,
+        @GetCurrentUser() user: CurrentUser,
+    ): Promise<PaginatedWarnings> {
+        // check if there are any scopes that the user does not have permission to
+        const allowedScopesForUser = user.permissions.find(({ permission }) => permission === "warnings")?.contentScopes;
```

Comment:
Is it possible that a user has the warnings permission multiple times? For instance, for some scopes by rule, and manually added for another scope. Then we should consider this here.

Reply:
I'm not sure. I tried to test it locally, but I can't impersonate and I'm unsure how to test it properly without it.

I found a message in our Slack that User Impersonation did not work in comet v8 but works since 8.0.0-beta.3 again. I probably need to rebase before I can check it. 

Is it okay if I check the topic again after the rebase or do you know how else I could test it?

Reply:
> Is it okay if I check the topic again after the rebase or do you know how else I could test it?

You can change the [AccessControlService](https://github.com/vivid-planet/comet/blob/feature/warning-module/demo/api/src/auth/access-control.service.ts) to not set all permissions for the user and manually assign it for two separate content scopes yourself.

Reply:
I assigned myself the warnings permission for main de and have this in the AccessControlService:

<img width="1112" alt="Screenshot 2025-04-25 at 14 51 46" src="https://github.com/user-attachments/assets/8627239c-7b0c-439b-9146-d6f6e28dba28" />


The user.permissions already return the "combined" contentScopes for the warnings
<img width="1003" alt="Screenshot 2025-04-25 at 14 52 52" src="https://github.com/user-attachments/assets/94347773-35e2-4b3f-85fc-6a3c3347c584" />

<img width="350" alt="Screenshot 2025-04-25 at 14 52 40" src="https://github.com/user-attachments/assets/0953f27c-ec4a-4a4b-a868-7899b0846c33" />